src/braket/ahs/atom_arrangement.py

```diff
@@ -0,0 +1,121 @@
+# language governing permissions and limitations under the License.
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from decimal import Decimal
+from enum import Enum
+from numbers import Number
+from typing import Iterator, List, Tuple, Union
+
+import numpy as np
+
+from braket.ahs.discretization_types import DiscretizationError, DiscretizationProperties
+
+
+class SiteType(Enum):
+    VACANT = "Vacant"
+    FILLED = "Filled"
+
+
+@dataclass
+class AtomArrangementItem:
+    coordinate: Tuple[Number, Number]
+    site_type: SiteType
+
+    def _validate_coordinate(self):
+        if len(self.coordinate) != 2:
+            raise ValueError(f"{self.coordinate} must be of length 2")
+        for idx, num in enumerate(self.coordinate):
+            if not isinstance(num, Number):
+                raise TypeError(f"{num} at position {idx} must be a number")
+
+    def _validate_site_type(self):
+        allowed_site_types = {SiteType.FILLED, SiteType.VACANT}
+        if self.site_type not in allowed_site_types:
+            raise ValueError(f"{self.site_type} must be one of {allowed_site_types}")
+
+    def __post_init__(self):
+        self._validate_coordinate()
+        self._validate_site_type()
+
+
+class AtomArrangement:
+    def __init__(self):
+        """Represents a set of coordinates that can be used as a register to an
+        AnalogHamiltonianSimulation.
+        """
+        self._sites = []
+
+    def add(
+        self,
+        coordinate: Union[Tuple[Number, Number], np.ndarray],
+        site_type: SiteType = SiteType.FILLED,
+    ) -> AtomArrangement:
+        """Add a coordinate to the atom arrangement.
+
+        Args:
+            coordinate (Union[Tuple[Number, Number], np.ndarray]): The coordinate of the
+                atom (in meters). The coordinates can be a numpy array of shape (2,)
+                or a tuple of int, float, Decimal
+            site_type (SiteType): The type of site. Optional. Default is FILLED.
+        Returns:
+            AtomArrangement: returns self (to allow for chaining).
+        """
+        self._sites.append(AtomArrangementItem(tuple(coordinate), site_type))
+        return self
+
+    def coordinate_list(self, coordinate_index: Number) -> List[Number]:
+        """Returns all the coordinates at the given index.
+
+        Args:
+            coordinate_index (Number): The index to get for each coordinate.
+
+        Returns:
+            List[Number]:The list of coordinates at the given index.
+
+        Example:
+            To get a list of all x-coordinates: coordinate_list(0)
+            To get a list of all y-coordinates: coordinate_list(1)
+        """
+        return [site.coordinate[coordinate_index] for site in self._sites]
+
+    def __iter__(self) -> Iterator:
+        return self._sites.__iter__()
+
+    def __len__(self):
+        return self._sites.__len__()
+
+    def discretize(self, properties: DiscretizationProperties) -> AtomArrangement:
```

Comment:
It's a little weird that the AHS class takes in a device object for discretize but the rest of these classes take in a new dataclass with a subset of the device data (the AHS class only appears to need this subset of data as well). Do we not expect the user to ever call discretize on these classes (AtomArrangement/Hamiltonian) ? Do we think that the inconsistency can be confusing ?

Reply:
It's the cleanest I could make this. 

It makes sense (to me) that the AnalogHamiltonianSimulation class understands devices, but I don't think the same should be for classes like TimeSeries. The cleanest break was to break out the actual data that was needed and feed it to the Hamiltonian and Registry classes.